Mathlib/NumberTheory/NumberField/Units.lean

```diff
@@ -502,6 +502,8 @@ theorem unitLattice_rank :
   rw [← Units.finrank_eq_rank]
   exact Zlattice.rank ℝ (unitLattice_span_eq_top K)
 
+set_option synthInstance.maxHeartbeats 27000
```

Comment:
This should only ever be use with `in` to attach it to a single declaration (and repeat as necessary)

Reply:
Done https://github.com/leanprover-community/mathlib4/pull/7060/commits/2631f67ad8dad4ca4be6f49769a21bc266303c5b thanks.